tools/icon_cutter/check.py

```diff
@@ -0,0 +1,57 @@
+import glob
+import os
+import re
+import sys
+import hashlib
+import platform
+import subprocess
+
+BUF_SIZE = 65536  # lets read stuff in 64kb chunks!
+
+def hash_file(path):
```

Comment:
don't hash the file, hash the actual contents of the png image data block. you'll get drifting over time from implementation differences wrt to pixel encoding (png supports 8 different ways of encoding the rgba values including palettes) and compression (the image data block is gzip compressed and different implementations or versions of the same implementations or just different oses or even env vars can cause it to compress at different gzip compression levels.).

use some python lib to convert the png to an array of 4byte values and hash that.

Reply:
excellent idea thank you.
what I'm doing now is using pillow to pull metadata, then pull the image data as a list of tuples and hash it as a string.

not sure if there's a better way of doing about that though

Reply:
worked out a way that's impervious to png dimension changes

Reply:
handled, we pull metadata and compare/output diffs (kinda ugly tho idk), then read the png as an image, split it into a flat array that doesn't care about dimensions, and then compares